### DISCUSSION TITLE
Help deploy hoppscotch Self-Host - Community Edition

### DISCUSSION THREAD

**Discussion by Xinuca** [hoppscotch/hoppscotch#4432] [Q&A] [answered]
Hello, I'm new here at hoppscotch.
I would like to implement hoppscotch Self-Host - Community Edition to work as a team.

But I'm having trouble, I've already tried different ways, the closest I can get is sending the email from the magic link but when I click it always goes back to the home page http://IP:3100

Could anyone help?

Installation data!

Guide - https://docs.hoppscotch.io/documentation/self-host/community-edition/getting-started

node -v
v22.9.0

npm -v
10.8.3

pnpm -v
9.12.1

docker -v
Docker version 26.1.4, build 5650f9b

git --version
git version 2.20.1

docker run -d --name postgres -e POSTGRES_USER=postgres -e POSTGRES_PASSWORD=hoppscotch -e POSTGRES_DB=hoppscotch -p 5432:5432 -v /opt/postgres:/var/lib/postgresql/data postgres:latest

docker pull hoppscotch/hoppscotch

docker run -p 3000:3000 -p 3100:3100 -p 3170:3170 --env-file .env --restart unless-stopped hoppscotch/hoppscotch

docker exec -it eb08114074e0 /bin/sh
Error response from daemon: Container eb08114074e0efcc65df35f419f6ba0318245c1ec86ca20575e697084b6ba0a9 is restarting, wait until the container is running

docker stop eb08114074e0
docker rm eb08114074e0

docker run -it --entrypoint sh --env-file .env hoppscotch/hoppscotch:latest
pnpx prisma migrate deploy

....
The following migration(s) have been applied:

migrations/
  └─ 20230406064219_init/
    └─ migration.sql
  └─ 20231106120154_embeds_addition/
    └─ migration.sql
  └─ 20231124104640_infra_config/
    └─ migration.sql
  └─ 20231130082054_collection_headers/
    └─ migration.sql
  └─ 20240226053141_full_text_search_additions/
    └─ migration.sql
  └─ 20240519093155_add_last_logged_on_to_user/
    └─ migration.sql
  └─ 20240520091033_personal_access_tokens/
    └─ migration.sql
  └─ 20240621062554_user_last_active_on/
    └─ migration.sql
  └─ 20240725043411_infra_config_encryption/
    └─ migration.sql
  └─ 20240726121956_infra_token/
    └─ migration.sql
    
exit

docker run -p 3000:3000 -p 3100:3100 -p 3170:3170 --env-file .env --restart unless-stopped hoppscotch/hoppscotch

Magic link -> http://10.0.10.157:3100/enter?token=cm26eqa8n0003131u6itqlssw

I click it always goes back to the home page http://10.0.10.157:3100

my .env

#-----------------------Backend Config------------------------------#
# Prisma Config
DATABASE_URL=postgresql://postgres:hoppscotch@10.0.10.157:5432/hoppscotch

# Auth Tokens Config
JWT_SECRET=secret1233
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY=3
REFRESH_TOKEN_VALIDITY=604800000
ACCESS_TOKEN_VALIDITY=86400000
SESSION_SECRET=anothersecretcode123

ALLOW_SECURE_COOKIES=true

# Sensitive Data Encryption Key while storing in Database (32 character)
DATA_ENCRYPTION_KEY=e7726a92f5803d8f7383aba0bbacd39d

# Hoppscotch App Domain Config
REDIRECT_URL=http://10.0.10.157:3000
WHITELISTED_ORIGINS=http://10.0.10.157:3170,http://10.0.10.157:3000,http://10.0.10.157:3100
VITE_ALLOWED_AUTH_PROVIDERS=GITHUB,EMAIL

# Github Auth Config
GITHUB_CLIENT_ID=myClientID
GITHUB_CLIENT_SECRET=myClientSecrets
GITHUB_CALLBACK_URL=http://10.0.10.157:3170/v1/auth/github/callback
GITHUB_SCOPE=user:email

# Mailer config
MAILER_SMTP_ENABLE=true
MAILER_USE_CUSTOM_CONFIGS=true
MAILER_ADDRESS_FROM=maria.stiedemann@ethereal.email

# The following are used if custom mailer configs is true
MAILER_SMTP_HOST=smtp.ethereal.email
MAILER_SMTP_PORT=587
MAILER_SMTP_SECURE=false
MAILER_SMTP_USER=maria.stiedemann@ethereal.email
MAILER_SMTP_PASSWORD=B7jKmDY4ydasvy3AMV
MAILER_TLS_REJECT_UNAUTHORIZED=true

# Rate Limit Config
RATE_LIMIT_TTL=60
RATE_LIMIT_MAX=100

#-----------------------Frontend Config------------------------------#

# Base URLs
VITE_BASE_URL=http://10.0.10.157:3000
VITE_SHORTCODE_BASE_URL=http://10.0.10.157:3000
VITE_ADMIN_URL=http://10.0.10.157:3100

# Backend URLs
VITE_BACKEND_GQL_URL=http://10.0.10.157:3170/graphql
VITE_BACKEND_WS_URL=ws://10.0.10.157:3170/graphql
VITE_BACKEND_API_URL=http://10.0.10.157:3170/v1

# Terms Of Service And Privacy Policy Links (Optional)
VITE_APP_TOS_LINK=https://docs.hoppscotch.io/support/terms
VITE_APP_PRIVACY_POLICY_LINK=https://docs.hoppscotch.io/support/privacy

# Set to `true` for subpath based access
ENABLE_SUBPATH_BASED_ACCESS=false

![Captura de ecrã de 2024-10-12 19-09-41](https://github.com/user-attachments/assets/91c4b729-6289-4060-9bc7-0f0c52523181)
![Captura de ecrã de 2024-10-12 19-10-05](https://github.com/user-attachments/assets/f0b1d4dc-caec-47dc-8ded-a65ec9e73456)
![Captura de ecrã de 2024-10-12 19-10-16](https://github.com/user-attachments/assets/9821eb88-647a-4150-8c8f-0c4d32430f7e)
![Captura de ecrã de 2024-10-12 19-10-27](https://github.com/user-attachments/assets/5f2fc783-9495-4e99-bf26-e4883b6be174)
![Captura de ecrã de 2024-10-12 19-10-48](https://github.com/user-attachments/assets/c077cfce-e1f0-4230-b239-071fb0012dc9)

**Comment by Xinuca** [comment]
even watching the step-by-step, I can't get it to work
https://www.youtube.com/watch?v=5CiI4PLfuJ0

**Comment by mirarifhasan** [comment, marked as answer]
Hi @Xinuca 
It seems you logged in using the magic link from the Admin Dashboard (running on IP:3100), which is why the email link directed you back there. 
To log in Hoppscotch App, use the Hoppscotch App interface (running on IP:3000). This way, the login email will provide a link directing you back to IP:3000.
